README.md

```diff
@@ -16,6 +16,12 @@ const xapi = jsxapi.connect('ssh://host.example.com', {
   password: 'password',
 });
 
+// Handle errors
+xapi.on('error', (err) => {
+  // !! Note of caution: This event might fire more than once !!
+  console.error(`xapi error: ${err}`);
+});
+
```

Comment:
Because the XAPI instance wraps not only the underlying transport, but also tons of semantic on top of the XAPI, I would say that this firing multiple times isn't necessarily an issue. I wouldn't trust we're at the point where we're 100% consistent on `error`'s being unrecoverable (if so, then it perhaps makes sense with an assertion on just a single `error` event).

Also, I guess we could envision in the future that we provide features like reconnect natively, which means error events propagate to the top, but the XAPI instance is still usable.

Reply:
Agreed. Native reconnect would very much make this more coherent.

Also agreed with regards to `error` events. They can come from many different sources, and I'm not so sure it is a good idea to guarantee single delivery, as that might end up removing error info in the wild.